### ISSUE TITLE
Need help

### ISSUE THREAD

**Issue by satansama6** [Unity-Technologies/ml-agents#195]
Hi. This is what i try to do:

![ezgif-4-cc136ff915](https://user-images.githubusercontent.com/30526720/34212775-02eccf8a-e5a6-11e7-8e50-12354e177700.gif)

I want to agent to move forward and pick up as many points as possible. 
STATES:
This is a tilemap system and I add 0 - nothing  1- wall 2 - point in the states, in a 10x10 square around the agent for a total of 100 states. 

REWARD:
0.005 if the agent moves forward
-0.0005 if the agent moves backward
2 if the agent picks up a Point

I trained for 2 million steps with these hyperparameters:

![](https://screenshots.firefoxusercontent.com/images/fee70177-34b5-4283-bed8-74d7d3cb34a5.png)

And as you can see, the agent is somewhat tries to move after the points, but not really at an acceptable rate. Does anyone know what should i change to make this work? Or I just need to let it train for like 10 million steps? Or the problem is with the tilemap approach and i got too many states for this to work? I appreciate every help.

Thank you!

**Comment by MarcoMeter** [comment]
Hi @satansama6 ,

the agent observes a grid of 100 tiles, right? Is this already the complete environment? In that case the agent would need to know its position. If the agent sees a fixed excerpt which is related to the agent's position, than this could be alright.

For sure you can still normalize your inputs. Ideally the values should be between -1 to +1 or 0 to +1. For nominal inputs, like your different kind of tiles can be normalized like this:
nothing = 0
wall = 0.5
point = 1

Another case for your input is concerned with the values, which you feed to the agent if its tile-based vision goes beyond a wall.  So if the agent is next to the wall, what does it see beyond the wall? I'd suggest to treat tiles which are out of bounds like a wall.

Did you define your brain to use discrete or continuous state space?

Concerning your rewards, I would suggest to use only  +1 for collection points as a start.


Concerning your hyperparameters, I'd give this a try:

batch_size = 16
hidden_units = 20

**Comment by satansama6** [comment]
I add 100 tiles around the player always as he moves. I don't know what you mean by defining discrete or continuous state space. I left it as default as continuous. I will apply the changes and report back. Thank you.

**Comment by MarcoMeter** [comment]
![statespace](https://user-images.githubusercontent.com/6996955/34216314-29362572-e5a8-11e7-999d-2dd80d624315.png)

I guess you've set it to continuous, which is totally fine.

**Comment by satansama6** [comment]
It seems that nothing changed. I get the same bad results. My question is:
Step: 1992500. Mean Reward: 38.67166666666607. Std of Reward: 17.61312641816379.
Step: 1995000. Mean Reward: 75.50500000000004. Std of Reward: 31.50000000000077.
Step: 1997500. Mean Reward: 60.00499999999993. Std of Reward: 7.257180352359291.
Step: 2000000. Mean Reward: 47.0049999999999. Std of Reward: 39.30224760324348.

These was the last steps and the rewards, but in game my cumulative reward is max 3. Is that normal? Isn't the mean reward the average of the cumulative rewards for the steps?

**Comment by vladimiroster** [comment]
Hi @satansama6, thanks for reaching out to us. Are you still having trouble?
